beacon_node/network/src/network_beacon_processor/gossip_methods.rs

```diff
@@ -172,23 +179,75 @@ impl<T: BeaconChainTypes> NetworkBeaconProcessor<T> {
     }
 
     pub(crate) fn handle_data_columns_to_publish(
-        &self,
+        self: &Arc<Self>,
         data_columns_to_publish: DataColumnsToPublish<T::EthSpec>,
+        block_root: Hash256,
     ) {
-        if let Some(data_columns_to_publish) = data_columns_to_publish {
-            self.send_network_message(NetworkMessage::Publish {
-                messages: data_columns_to_publish
-                    .iter()
-                    .map(|d| {
-                        let subnet = DataColumnSubnetId::from_column_index::<T::EthSpec>(
-                            d.index as usize,
-                            &self.chain.spec,
+        let Some(mut data_columns_to_publish) = data_columns_to_publish else {
+            return;
+        };
+        // let self_executor= self.clone();
+        let self_clone = self.clone();
+        self.executor.spawn(
+            async move {
+                let is_supernode = self_clone
+                    .chain
+                    .data_availability_checker
+                    .get_custody_columns_count()
+                    == self_clone.chain.spec.number_of_columns;
+
+                let publish_fn = |columns: DataColumnSidecarList<T::EthSpec>| {
+                    self_clone.send_network_message(NetworkMessage::Publish {
+                        messages: columns
+                            .into_iter()
+                            .map(|d| {
+                                let subnet = DataColumnSubnetId::from_column_index::<T::EthSpec>(
+                                    d.index as usize,
+                                    &self_clone.chain.spec,
+                                );
+                                PubsubMessage::DataColumnSidecar(Box::new((subnet, d)))
+                            })
+                            .collect(),
+                    });
+                };
+
+                if !is_supernode {
+                    publish_fn(data_columns_to_publish);
```

Comment:
do you meant to just return here?

Reply:
oh yeah I guess we could, previously we were publishing there for non-supernodes though, I think

Reply:
This should be unreachable because `data_columns_to_publish` passed to this function would be `None` as full node never perform reconstruction